### ISSUE TITLE
Gitlab CI passing green when cypress can't connect to the url

### ISSUE THREAD

**Issue by avatsaev** [cypress-io/cypress#1063]
This is docker based CI with cypress/base docker image

### Is this a Feature or Bug?

Gitlab job succeeds even when url reports 404 error

```
Opening Cypress...
  (Tests Starting)


  
GET / 404 12.412 ms - 13

    1) "before each" hook for "should display header logo"


  0 passing (330ms)
  1 failing

  1)  "before each" hook for "should display header logo":
     CypressError: cy.visit() failed trying to load:

http://localhost:8080/

The response we received from your web server was:

  > 404: Not Found

This was considered a failure because the status code was not '2xx'.

If you do not want status codes to cause failures pass the option: 'failOnStatusCode: false'

Because this error occurred during a 'before each' hook we are skipping all of the remaining tests.
      at Object.cypressErr (http://localhost:38719/__cypress/runner/cypress_runner.js:65873:11)
      at Object.throwErr (http://localhost:38719/__cypress/runner/cypress_runner.js:65838:18)
      at Object.throwErrByPath (http://localhost:38719/__cypress/runner/cypress_runner.js:65865:17)
      at http://localhost:38719/__cypress/runner/cypress_runner.js:57481:31
      at visitFailedByErr (http://localhost:38719/__cypress/runner/cypress_runner.js:57107:12)
      at http://localhost:38719/__cypress/runner/cypress_runner.js:57462:22
      at tryCatcher (http://localhost:38719/__cypress/runner/cypress_runner.js:6268:23)
      at Promise._settlePromiseFromHandler (http://localhost:38719/__cypress/runner/cypress_runner.js:4290:31)
      at Promise._settlePromise (http://localhost:38719/__cypress/runner/cypress_runner.js:4347:18)
      at Promise._settlePromise0 (http://localhost:38719/__cypress/runner/cypress_runner.js:4392:10)
      at Promise._settlePromises (http://localhost:38719/__cypress/runner/cypress_runner.js:4467:18)
      at Async._drainQueue (http://localhost:38719/__cypress/runner/cypress_runner.js:1200:16)
      at Async._drainQueues (http://localhost:38719/__cypress/runner/cypress_runner.js:1210:10)
      at Async.drainQueues (http://localhost:38719/__cypress/runner/cypress_runner.js:1084:14)




  (Tests Finished)

  - Tests:           0
  - Passes:          0
  - Failures:        0
  - Pending:         0
  - Duration:        0 seconds
  - Screenshots:     0
  - Video Recorded:  false
  - Cypress Version: 1.1.4


  (All Done)
npm info ok 
Job succeeded

```

### Desired behavior:

The job must fail when cypress can't connect to the http server

Cypress ver: 1.1.4

Reporter: mochawesome

**Comment by bahmutov** [comment]
I will investigate using our GitLabCI example https://gitlab.com/cypress-io/cypress-example-docker-gitlab

**Comment by brian-mann** [comment]
@avatsaev and @bahmutov no need to investigate related to CI. This does not appear related to CI or containers at all.

How do I know?

Because the summary shows `zero` failures - even though we clearly picked up the failure when the test failed.

This should be reproducible in any environment on any system. @avatsaev we would really need a minimal reproducible amount of test code that causes this. Can you remove every single line of test code until the bare minimum to reproduce this?

The juggle of test failures to exit codes is one of the most tested places in Cypress - and its unlikely this is simple to reproduce on our end from the information you've provided. I'm guessing there is something that makes your situation unique which causes this to happen.

**Comment by avatsaev** [comment]
I only have one test, and I've used mochawesome reporter from the docs here: https://docs.cypress.io/guides/guides/reporters.html#Local-Reporters

**Comment by brian-mann** [comment]
I didn't realize mochaawesome was the source of the problem. We'll get it fixed.

**Comment by brian-mann** [comment]
After doing some research the issue appears to be more complicated. `mochawesome` does not correctly hold the number of `failures` on its `stats` property - which is what we use to denote exit code unfortunately.

I'm going to have to dig into their source code to see where they handle exit code and why it's different. A quick test shows me that when failures occur in `beforeEach` hooks they do not consider that a failure - which is why this is happening to you.

**Comment by brian-mann** [comment]
Released in [`1.4.2`](http://localhost:2222/guides/references/changelog.html#1-4-2).
